US-1.8: Review Event Creation

RGPosadas opened the pull request:
This PR implements #48.

To test, simply create an event - it should show an event preview where you can choose to edit or submit your event form.

ritchellesap committed:
refactor: add children prop to Mull button #48

Co-authored-by: Cristian Aldea <aldea.cristian@outlook.com>

ritchellesap committed:
connect create event form with preview event page #48

Co-authored-by: Cristian Aldea <aldea.cristian@outlook.com>

ritchellesap committed:
add onButtonClick prop to EventPage #48

Co-authored-by: Cristian Aldea <aldea.cristian@outlook.com>

ritchellesap committed:
refactor: remove console log line #48

ritchellesap committed:
update snapshots #48

ritchellesap committed:
add new event page props to unit tests #48

ritchellesap committed:
refactor: cleanup implemented to-do's #48

ritchellesap committed:
refactor: move frame sizes into exportable const for reusability across e2e tests #48

ritchellesap committed:
add US-1.8 e2e test #48

RGPosadas commented:
Just realized that the E2E test won't pass since I hardcoded the expected date. Fixing this now.

RGPosadas commented:
Fixed it now. This PR is ready for review again 😄

r-saba commented on apps/mull-ui/src/app/components/event-page-header/event-page-header.tsx:
```diff
@@ -31,14 +38,9 @@ export const EventPageHeader = ({ event, prevPage }: EventPageHeaderProps) => {
   return (
     <div className="event-page-header">
-      <MullBackButton>{prevPage}</MullBackButton>
+      <MullBackButton onClick={onBackButtonClick}>{prevPage}</MullBackButton>
```
This doesn't read so well `onClick={onBackButtonClick}`. I usually see something like `onClick={handleBackButton}`. Curious what you think

RGPosadas replied:
If that is what's most common, then I'll rename it accordingly 😄

r-saba commented on apps/mull-ui/src/app/pages/create-event/create-event.tsx:
```diff
@@ -53,6 +54,8 @@ const CreateEventPage = ({ history }: CreateEventProps) => {
   // Uploaded Image File
   const [imageURLFile, setImageURLFile] = useState<string>(null); // Path of uploaded image on client, to be used in image previews
   const [file, setFile] = useState<File>(null); // Uploaded image file blob
+  const [review, setReview] = useState<boolean>(false); // Show either form or review page
```
maybe `inReview` or `isInReview` to make it clear it's a flag?

r-saba commented on apps/mull-ui/src/app/components/event-page-info/event-page-info.tsx:
```diff
@@ -17,11 +17,12 @@ import { ExpandableText } from './../expandable-text/expandable-text';
-  event: IEvent;
+  event: Partial<IEvent>;
   className?: string;
+  onButtonClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
```
Can this be more descriptive as to what it does? `handleEventSubmission` or whatever you think is appropriate

RGPosadas replied:
Sure, something like `handleMullButton` since the button will be used to create events in review mode, and join events in preview mode. Will also add in a comment to explain why it's somewhat an abstracted name.

r-saba reviewed (dismissed):
LGTM! No blockers.
Found some unused imports in `app.tsx` just double check if there's any other by running the linter. 

Somewhat unrelated but I see a lot of room for refactoring in `create-event.tsx`

ritchellesap committed:
refactor: rename props #48
remove unused import in app.tsx
rename state variable

RGPosadas commented:
> LGTM! No blockers.
> Found some unused imports in `app.tsx` just double check if there's any other by running the linter.
> 
> Somewhat unrelated but I see a lot of room for refactoring in `create-event.tsx`

@r-saba 
Thanks for the reminder about linting! Addressed your comments in this commit: 708fc71 😄

ys-lin reviewed (dismissed):
LGTM 👍 

- [x] User Acceptance Flow is respected
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every component and function
- [x] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2000%3A222919)

r-saba commented on apps/mull-ui/src/app/components/event-page-info/event-page-info.tsx:
```diff
@@ -17,11 +17,13 @@ import { ExpandableText } from './../expandable-text/expandable-text';
 }
 
-export const EventPageInfo = ({ event, className = '' }: EventPageInfoProps) => {
+export const EventPageInfo = ({ event, className = '', handleMullButton }: EventPageInfoProps) => {
```
Hey @RGPosadas did you know you can document your component like so. This way when someone hovers over the component to figure out what `handleMullButton` does it will show them the text. This way we won't need the comment above as well
```suggestion
/**
 * @param handleMullButton Function called when button at bottom of event-page-info is pressed
 */
export const EventPageInfo = ({ event, className = '', handleMullButton }: EventPageInfoProps) => {
```

RGPosadas replied:
Ooh I see, thank you for the fyi!

JimLau49 reviewed (dismissed):
Nice work Richelle 👍  not really anything i want addressed in this ticket that wasn't handled in the event card ticket. 
- [x] User Acceptance Flow is respected
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every component and function
- [x] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2000%3A222919)

RGPosadas commented:
Sorry for staling your reviews @JimLau49 @ys-lin, I used jsdoc instead of a regular comment like @r-saba suggested!
That's the only change I've added.

r-saba reviewed (approved):
LGTM! Good work!
- [x] User Acceptance Flow is respected
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every component and function
- [x] There must exist an automated Cypress acceptance test which tests the user acceptance flow
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2000%3A222919)
